File: QuoteCalculator/QuoteCalculator.Domain/Entities/LoanApplication.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace QuoteCalculator.Domain.Entities
{
    public class LoanApplication
    {
        public int LoanApplicationId { get; set; }
        public Guid LoanApplicationPublicId { get; set; }
        public int ApplicantId { get; set; }
        public Applicant Applicant { get; set; } = default!;
        public decimal Amount { get; set; }
        public int Term { get; set; }
        public int ProductType { get; set; }
        public int Status { get; set; }
        public decimal EstablishmentFee { get; set; }
        public decimal InterestFee { get; set; }
        public decimal WeeklyRepayment { get; set; }
        public decimal MonthlyRepayment { get; set; }
        public decimal TotalRepayment { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public string? ReferenceNumber { get; set; }
        public string? Remarks { get; set; }
        public uint RowVersion { get; set; }
    }
}

File: quote-calculator-app/src/types/LoanApplication.ts
export interface LoanApplication {
    id: string;
    firstName: string;
    lastName: string;
    email: string;
    amount: number;
    term: number;
}

export interface LoanApplicationDetails {
    loanApplicationPublicId: string;
    applicantId: number;
    title: string;
    firstName: string;
    lastName: string;
    email: string;
    mobileNumber: string;
    dateOfBirth: Date;
    amount: number;
    term: number;
    productType: number;
}

export interface CalculateLoanRequest {
    productTypeId: number | undefined;
    amount: number | undefined;
    term: number | undefined;
}

export interface DraftLoanApplicationRequest {
    loanApplicationPublicId: string;
    applicantId: number;
    title: string;
    firstName: string;
    lastName: string;
    email: string;
    mobileNumber: string;
    amount: number;
    term: number;
    productTypeId: number;
}

export interface CalculateLoanDetails {
    amount: number;
    term: number;
    establishmentFee: number;
    interestFee: number;
    weeklyRepayment: number;
    monthlyRepayment: number;
    totalRepayment: number;
}

export interface FinalizeLoanApplicationRequest {
    loanApplicationPublicId: string;
    applicantId: number;
    firstName: string;
    lastName: string;
    email: string;
    mobileNumber: string;
    amount: number;
    term: number;
    interestFee: number;
    weeklyRepayment: number;
    monthlyRepayment: number;
    totalRepayment: number;
}

export interface ApplicationReceiptDetails {
    firstName: string;
    email: string;
    amount: number;
    referenceNumber: string;
    status: number;
}